controllers/space/space_controller.go

```diff
@@ -0,0 +1,343 @@
+		builder = builder.Watches(source.NewKindWithCache(&toolchainv1alpha1.NSTemplateSet{}, memberCluster.Cache),
+			&handler.EnqueueRequestForObject{},
+		)
+	}
+
+	return builder.Complete(&Reconciler{
+		Client:         mgr.GetClient(),
+		MemberClusters: memberClusters,
+	})
+}
+
+//+kubebuilder:rbac:groups=toolchain.dev.openshift.com,resources=spaces,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=toolchain.dev.openshift.com,resources=spaces/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=toolchain.dev.openshift.com,resources=spaces/finalizers,verbs=update
+
+// Reconcile ensures that there is an NSTemplateSet resource defined in the target member cluster
+func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+	logger.Info("reconciling Space")
+
+	// Fetch the Space
+	space := &toolchainv1alpha1.Space{}
+	err := r.Client.Get(context.TODO(), request.NamespacedName, space)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, errs.Wrap(err, "unable to get the current Space")
+	}
+	if !util.IsBeingDeleted(space) {
+		// Add the finalizer if it is not present
+		if err := r.addFinalizer(logger, space); err != nil {
+			return reconcile.Result{}, err
+		}
+	} else {
+		requeue, err := r.ensureSpaceDeletion(logger, space)
+		return reconcile.Result{Requeue: requeue}, err
+	}
+	// ensure that there's a NSTemplateSet on the Target Cluster
+	// will trigger a requeue until the NSTemplateSet exists and is ready,
+	// so the Space can be in `ready` status as well
+	requeue, err := r.ensureNSTemplateSet(logger, space)
+	return ctrl.Result{Requeue: requeue}, err
+}
+
+// setFinalizers sets the finalizers for Space
+func (r *Reconciler) addFinalizer(logger logr.Logger, space *toolchainv1alpha1.Space) error {
+	// Add the finalizer if it is not present
+	if !util.HasFinalizer(space, toolchainv1alpha1.FinalizerName) {
+		logger.Info("adding finalizer on Space")
+		util.AddFinalizer(space, toolchainv1alpha1.FinalizerName)
+		if err := r.Client.Update(context.TODO(), space); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// ensureNSTemplateSet creates the NSTemplateSet on the target member cluster if it does not exist,
+// and updates the space's status accordingly.
+// returns `true` after creating the NSTemplateSet and until it is `ready`
+func (r *Reconciler) ensureNSTemplateSet(logger logr.Logger, space *toolchainv1alpha1.Space) (bool, error) {
+	if space.Spec.TargetCluster == "" {
+		return false, fmt.Errorf("missing target member cluster for Space '%s'", space.Name)
+	}
```

Comment:
This is a valid state - there is gonna be a completion controller that will set the `targetCluster` based on the capacity of the member clusters: https://issues.redhat.com/browse/CRT-1333
so instead of returning an error, we should just set the state that it's not ready because there is no targetCluster set (for example because of the capacity issues)

Reply:
I believe this was addressed in 1285823 (even though I missed your comment here)